tools/clang/lib/AST/Type.cpp

```diff
@@ -2497,13 +2497,13 @@ StringRef BuiltinType::getName(const PrintingPolicy &Policy) const {
   case Char_S:            return "char";
   case Char_U:            return "char";
   case SChar:             return "signed char";
-  case Short:             return "short";
+  case Short:             return /* "short" */ "int16_t" /* HLSL Change */;
   case Int:               return "int";
   case Long:              return "long";
   case LongLong:          return "long long";
   case Int128:            return "__int128";
   case UChar:             return "unsigned char";
-  case UShort:            return "unsigned short";
+  case UShort:            return /* "unsigned short" */ "uint16_t" /* HLSL Change */;
   case UInt:              return "unsigned int";
```

Comment:
What about some of the neighboring types that also don't match HLSL names?  Shouldn't we update them as well?  Like:
```
long -> int (do we hit this case?)
long long -> uint64_t
unsigned int -> uint (do we hit this case?)
unsigned long -> uint
unsigned long long -> uint64_t
```

Reply:
Added those too. Strangely `uint16_t` and `uint64_t` are unique basic types, but `uint` is just a typedef of `unsigned int`.